### PR TITLE
[OpenCage] Locality

### PR DESCRIPTION
**Pull request by nicolassing**
use city, town village or hamlet as locality

### CODE REVIEW

**Comment by nicolassing** [comment]
poke @mtmail

If it's ok for you, I'll add some tests

**Comment by mtmail** [comment]
Sure. Thanks.

Fyi, This is a list of aliases we use for our address formatting. https://github.com/OpenCageData/address-formatting/blob/master/conf/components.yaml

**Comment by nicolassing** [comment]
ok thx :)

**Comment by willdurand** [comment]
poke @geocoder-php/geocoder

File: src/Geocoder/Provider/OpenCage.php [hunk 1]
```diff
@@ -138,21 +138,74 @@ private function executeQuery($query)
...
+    /**
+     * @param array $components
+     * @param array $keys
+     *
+     * @return null|string
+     */
+    protected function guessBestComponent(array $components, array $keys)
+    {
+        foreach ($keys as $key) {
+            if (isset($components[$key]) && !empty($components[$key])) {
```
Comment (Baachi) [inline src/Geocoder/Provider/OpenCage.php]: the `isset` check is superfluously.

**Comment by Baachi** [comment]
Looks good :+1:

**Comment by toin0u** [comment]
Small CS stuffs but overall :+1:

**Comment by willdurand** [comment]
thanks @nicolassing

**Pull request merged**